### ISSUE TITLE
New release?

### ISSUE THREAD

**Issue by bluescarni** [shibatch/sleef#442]
Hello,

First of all, thanks for this great library!

Is there an ETA for the release of version 3.6? Apologies if this is the wrong communication channel for this sort of question...

**Comment by shibatch** [comment]
It is difficult to spend any more time on this activity because my 
institute does not appreciate SLEEF project AT ALL.

**Comment by bluescarni** [comment]
I am really sorry to hear that :(

**Comment by carlkl** [comment]
There is no other OSS implementation of trigonometric functions that can fulfill the triad of performance, accuracy, and platform independence in the way Sleef does. It would be very sad if Sleef just fades away.

**Comment by shibatch** [comment]
Before version 2.80, SLEEF has been developed intermittently. The way 
SLEEF is developed is just reverted to that state.

As you know, this is a fundamental problem of OSS development. Usually, 
development of software like SLEEF would require several 10 times of the 
amount I received as donations so far. It is impossible to continue the 
pace of development with that amount.

**Comment by carlkl** [comment]
Maybe Sleef development needs something like that: https://chanzuckerberg.com/eoss/

NumPy and OpenBLAS as an example:
https://chanzuckerberg.com/eoss/proposals/strengthening-numpys-foundations-growing-beyond-code/

**Comment by shibatch** [comment]
The original plan was to gain recognition from my institute for 
maintaining software that has a significant impact on society.
I succeeded in publishing a good paper on SLEEF and received donations 
from large companies, so I thought that people in my institute would 
understand the importance of SLEEF.
In reality, however, i found that people in my institute do not 
distinguish between SLEEF and some silly proof-of-concept codes for 
student researches.
They frequently tell us that it is important to implement the results of 
the researches to directly contribute to the society, but it turned out 
that they simply say it like that and have no interest in actually doing it.

**Comment by carlkl** [comment]
Sad to hear. However, it makes no sense to comment on the research policy of an institute, it's on there own. If maintenance work is not supported (because it is does not qualify as research work?) further 3rd party donations may be the answer. Keeping up the motivation for a bigger OSS project can be quite tedious.

BTW: In the last two years I tried to promote Sleef here and there in the Python/Numpy community, but there was not much more than some courteous interest. And I didn't yet found time to get my hands dirty to integrate into a python project worth mentioning (last year I started with Sleef and https://github.com/pydata/numexpr but didn't go far). At least I can say a lot of developers in this community are aware of Sleef and know that this is a very interesting library when it comes to SIMD vectorization. On the other hand pytorch is the only python project I'm aware of that is using Sleef in production.

Numpy decided to use to integrate the OSS part of SVML into its codebase (https://github.com/numpy/SVML). This works only for avx-512. And some parts of numpy are now written with universal intrinsics: see https://numpy.org/devdocs/reference/simd/index.html and https://github.com/numpy/numpy/blob/main/numpy/core/src/umath/loops_trigonometric.dispatch.c.src, but Sleef was never considered to integrate in into Numpy. In comparison: Sleef is feature complete (C99 trig functions) and could yield better results, especially for the full numeric range IMHO.

Maybe a new Sleef release and some more promotion could help to gain interest in this community and for gain a grant.(Numfocus i.e.)

Just my two cents.

**Comment by shibatch** [comment]
I think that's a kind of idealism, and in reality, everyone cares about 
their own career and salary. If you turn your back on that, people will 
eventually leave OSS development. Even the GNU project grew because of 
the support of universities.

Looking at the current situation, it is very difficult to make a living 
only by developing OSS.
The fact that institutes have thrown away their development support 
basically means that the developers have to use their spare time to 
maintain the software.

Translated with www.DeepL.com/Translator (free version)

**Comment by carlkl** [comment]
Thanks for this insight. I faced the same situation:  OSS work in spare time means that good ideas come to a premature end due to slow progress. And small grants don't really help in the end.

**Comment by blapie** [comment]
The 3.6 release was tagged today.
I am probably going to close this thread by virtue of that.
The discussion is very interesting, so I will probably re-open it as a regular discussion as opposed to an issue.
That should help us reflect on the new type of maintainer-ship we are currently experimenting.
